tests/features/S2N_LIBCRYPTO_SUPPORTS_RSA_PSS_SIGNING.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <openssl/rsa.h>
+
+/* OpenSSL 1.1.1d 10 Sep 2019 is broken, so disable on that version. For further info see: crypto/evp/p_lib.c:469
+ *
+ * This feature requires this Openssl commit for Openssl 1.1.x versions: openssl/openssl@4088b92
+ */
+#if OPENSSL_VERSION_NUMBER <= 0x1010104fL
```

Comment:
What happens if OPENSSL_VERSION_NUMBER is not defined (as in, we are not using openssl as a libcrypto)?

Reply:
Locally, this did not change the feature flag, but I'm also not sure how/where this is defined during the CI. My IDE traces it to the built-in openssl library definition of it

I also noticed that try-compiling `EVP_PKEY_CTX_set_rsa_pss_saltlen(NULL, NULL);` turns the flag to FALSE while try-compiling `RSA_get0_pss_params(NULL);` turns the flag to TRUE (built with aws_lc)

Reply:
Might be worth wrapping the whole thing just to be sure

```c
#if defined(OPENSSL_VERSION_NUMBER)
#if OPENSSL_VERSION_NUMBER <= 0x1010104fL
#error "Version of OpenSSL does not support RSA-PSS"
#endif
#endif
```

Reply:
Also, keep in mind that different functions require different include statements. So if you wanted to use EVP_PKEY_CTX_set_rsa_pss_saltlen, the include path is <openssl/evp.h> and if you wanted to use RSA_get0_pss_params, the include path would be <openssl/rsa.h>. 

Not sure if that was the issue you were hitting, but those functions are both supported by aws-lc, so it's weird that they would evaluate to different try-compile results.

Reply:
I think that was the issue. When I used <openssl/evp.h> as import the flag now turns to be TRUE. But why is this the case? I see a header definition of `EVP_PKEY_CTX_set_rsa_pss_saltlen` inside `openssl/rsa.h`, but not inside my `openssl/evp.h`